本番環境（prod）の変更をCI/CD経由に限定する運用制御

**Related:** #101

## 概要

本番環境（prod）の Terraform 変更をローカル実行から禁止し、CI/CD パイプライン経由のみで apply/destroy 可能にする。  
誤操作防止・監査性・再現性を担保する。

## 配置管理戦略

**「ローカルは .env、本番は CI Secrets」が一番お行儀いいです。**

- **ローカル開発環境**: `.env` で管理（DEV/PROD切り替え用、ローカル実行専用）
- **本番環境（PROD）**: GitHub Secrets で厳密に管理（CI/CD実行専用）
  - `HCP_TERRAFORM_TOKEN`, `PROD_SNOWFLAKE_*` など機密情報は絶対に `.env` に置かない
  - ローカルからのPROD実行を物理的に防止
- **境界制御**: CI環境変数の存在で自動判定（`GITHUB_ACTIONS=true` 等）

## 背景

APP_ENV による dev/prod 切り替えは可能だが、運用ポリシーとして prod 変更は CI/CD のみ許可したい。

## 目標

- [ ] prod への Terraform 変更は CI/CD 経由のみ許可
- [ ] ローカル端末からの prod apply/destroy は明示的に失敗
- [ ] 本番変更の実行履歴を CI/CD ログで追跡可能にする
- [ ] エラーメッセージで「prod変更はCI/CDのみ」を明示的に通知

## 実装方針

### 1. 実行経路の制御

- prod workspace の apply 権限を CI/CD サービスアカウントに限定
- 開発者ローカル認証情報では prod workspace へ apply 不可

### 2. ラッパースクリプトのガード

- `APP_ENV=prod` かつ CI 環境変数未検出時は plan 以外を拒否
- エラーメッセージで「prod変更はCI/CDのみ」を明示

### 3. CI/CD 側の統制

- main ブランチマージで prod plan/apply を実行
- 必要に応じて手動承認ステップを追加（GitHub Environments）
- 実行ログと差分を保存

## タスク

- [ ] prod workspace の実行権限を CI/CD 中心に再設計
- [ ] ラッパーに「prod ローカル apply/destroy 禁止」ガード実装
- [ ] CI/CD ジョブに prod plan/apply フロー実装
- [ ] README に「prodはCI/CDのみ」の運用ルールを明記
- [ ] ローカルから prod apply/destroy が失敗することを確認
- [ ] CI/CD から prod apply が成功することを確認

## 期待される動作

```bash
# ⛔️ ローカルでは拒否される
$ APP_ENV=prod ./terraform/tf apply
[ERROR] 本番環境（prod）への apply/destroy はローカルから実行できません。
        必ず CI/CD パイプライン経由でのみ実行可能です。

# ✅ 本番planは読み取り専用で実行可能
$ APP_ENV=prod ./terraform/tf plan
(計画出力)

# ✅ CI/CD環境では正常に実行
$ CI=true APP_ENV=prod terraform apply
(apply実行)
```

## 受け入れ条件

- [ ] ローカル実行: prod apply/destroy は常にブロック
- [ ] CI/CD 実行: prod plan/apply が正常完了
- [ ] 監査: 実行主体・時刻・差分が追跡可能
- [ ] ドキュメント: 運用ルールが明文化されチーム合意済み

## 参考資料

- セキュリティベストプラクティス: 環境分離、最小権限の原則
- CI/CD ワークフロー: 承認ゲート、監査ログ

--- a/terraform/README.md
+++ b/terraform/README.md
@@ -353,7 +353,9 @@ CI/CD ワークフローで使用する秘密情報は、GitHub Repository Secre
 
 ## 2. 実行手順
 
-Docker コンテナ内で実行する場合も、初回は `terraform login` による認可が必要です。
+`terraform/tf` は `.env.shared` / `.env` または CI Secrets から
+`TF_TOKEN_app_terraform_io`（互換: `HCP_TERRAFORM_TOKEN`, `HCP_TF_TEAM_API_TOKEN`）を読み込みます。
+`terraform login` は不要です。
 
 ### 2.1. DEV / PROD 共通手順
 
@@ -381,6 +383,33 @@ Docker コンテナ内で実行する場合も、初回は `terraform login` に
 
 - CI では `APP_ENV=prod ./terraform/tf plan` / `apply` の形式を利用できます。
 - ラッパーを使わずに `terraform` コマンドを直接実行する運用は推奨しません。
+
+## 2.2. 復旧手順（運用者向け Runbook）
+
+prod apply が失敗した場合、次の順で切り分けて復旧してください。
+
+1. **Actions 実行履歴の確認**
+    - 対象 run: `CI Pipeline` の `terraform-prod-apply`
+    - `prod-apply.log` artifact をダウンロードしてエラー行を特定
+2. **Environment 保護の確認**
+    - `Settings > Environments > prod`
+    - Deployment branches が `main` 限定か確認
+    - Required reviewers が有効か確認
+3. **Secrets の有効性確認**
+    - `HCP_TERRAFORM_TOKEN` が CI 用 Team API Token であること
+    - `HCP_TF_ORGANIZATION` が正しいこと
+    - トークン失効時は再発行して Secrets を更新
+4. **HCP Terraform 権限確認**
+    - CI 用 Team が prod workspace で apply 可能であること
+    - 開発者ロールに prod apply 権限が付与されていないこと
+5. **再実行**
+    - 必要に応じて main へ修正コミットをマージし、再度 approval 後に apply
+
+**緊急時（break-glass）ポリシー:**
+
+- 原則としてローカルから prod 変更は行わない
+- 例外対応は Issue を起票し、実施者・理由・実行コマンド・結果を記録する
+- 事後に CI 経路へ設定を戻し、Secrets と権限の棚卸しを実施する
 
 ## 3. 設計判断（ADR）
 

